pennylane/transforms/optimization/cancel_inverses.py

```diff
@@ -114,32 +114,31 @@ def qfunc(x, y, z):
             # If the wires are the same, then we can safely remove both
             if current_gate.wires == next_gate.wires:
                 list_copy.pop(next_gate_idx + 1)
-            # If wires are not equal, there are two things that can happen
+            # If wires are not equal, there are two things that can happen.
```

Comment:
Yes, this change reduces the maximal block indentation level from 6 to 5, i.e. to the "allowed" maximum.
Based on the code duplication, it feels like this can be simplified/restructured further. Essentially there only ever happen two things in the leafs of the nested if-else clauses.

Reply:
By moving the final `list_copy.pop(0)` further up, I made use of `continue` statements a bit more and (imho) improved the readability a bit :)